src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -773,6 +773,19 @@ public JsonOperationContext GetContextForNewCompareExchangeValue()
                 return _context;
             }
 
+            public bool MaybeFlush()
+            {
+                if (_clusterTransactionCommands.Length == 0)
+                    return false;
+                return ++_maybeFlushTestCount > 1024;
```

Comment:
I think we should at least use the same number as in BatchSize? 10k IIRC

Reply:
AbstractDatabaseCompareExchangeActions.BatchSize

Reply:
Actually, after thinking about that, higher number would be better, but I'm wondering if in this case the time constraint would not be better? You do not want to keep read tx opened for too long, but it does not mean that you want to spam commands here with few items. Wouldn't it be better to switch to e.g. 30s instead?

Reply:
The best option would be to do that _before_ we submit the next documents batch, to be honest.
The problem is that this is a bigger change.